src/Host.cpp

```diff
@@ -3305,7 +3319,9 @@ std::pair<bool, QString> Host::setWindow(const QString& windowname, const QStrin
     auto pD = mpConsole->mDockWidgetMap.value(windowname);
     auto pSW = mpConsole->mScrollBoxMap.value(windowname);
 
-    if (!pSW && !pD && windowname.toLower() != QLatin1String("main")) {
+    if (!pSW && !pD && windowname.compare(QLatin1String("main"), Qt::CaseInsensitive)) {
+        // Third argument is non-zero (i.e. true) if the window name is NOT
```

Comment:
I'm not sure what this means

Reply:
It is a bit of an anti-pattern to use `toUpper()` or `toLower()` to then do a comparison to a fixed string when the `QLatin1String(...)` class has a `compare(...)` method and that has a built-in case sensitivity option (see "**30 qstring-insensitive-allocation**" in **[Uncovering 32 Qt Qt best practices at compile time with Clazy](https://www.kdab.com/uncovering-32-qt-best-practices-compile-time-clazy/)** ) - however the logic of the return value `0` (i.e. the C equivalent of `false`) for a match is a tiny bit counter intuitive.

Reply:
Oh. The comment does not reflect this I don't think, it's confusing me more than helping...